src/lib/common/sol-power-supply-impl-linux.c

```diff
@@ -181,37 +181,33 @@ _get_enum_prop(const char *name, enum sol_power_supply_prop prop,
     return 0;
 }
 
-static bool
+static enum sol_util_iterate_dir_reason
 _list_supplies_cb(void *data, const char *dir_path, struct dirent *ent)
 {
     struct list_data *ldata = data;
     enum sol_power_supply_type type;
     char *name;
     int r;
 
-    if ((!strcmp(ent->d_name, ".")) || (!strcmp(ent->d_name, ".."))) {
-        return false;
-    }
-
     if (ldata->check_type) {
         r = sol_power_supply_get_type(ent->d_name, &type);
-        SOL_INT_CHECK(r, < 0, false);
+        SOL_INT_CHECK(r, < 0, SOL_UTIL_ITERATE_DIR_CONTINUE);
 
         if (type != ldata->type)
-            return false;
+            return SOL_UTIL_ITERATE_DIR_CONTINUE;
     }
 
     name = strdup(ent->d_name);
-    SOL_NULL_CHECK(name, false);
+    SOL_NULL_CHECK(name, -ENOMEM);
 
     r = sol_ptr_vector_append(ldata->list, name);
     if (r < 0) {
         SOL_WRN("Failed to append supply to list %s", name);
         free(name);
-        return false;
+        return r;
```

Comment:
On 27-04-2016 11:14, Gustavo Lima Chaves wrote:
> In src/lib/common/sol-power-supply-impl-linux.c
> <https://github.com/solettaproject/soletta/pull/1940#discussion_r61264234>:
>
>>
>>      r = sol_ptr_vector_append(ldata->list, name);
>>      if (r < 0) {
>>          SOL_WRN("Failed to append supply to list %s", name);
>>          free(name);
>> -        return false;
>> +        return r;
>
> That's my point -- under memory shortage there's no point in continuing

Fair.

>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1940/files/6d8331b377358a901201bc6c10b1e969c7f5328c..50301098ec7a9853d9d82e7b84e562e69ea2709d#r61264234>
>